crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -438,14 +447,31 @@ async fn wait_for_signal() -> Result<i32> {
     }
 }
 
+/// The pooling allocator is tailor made for the `wasi/http` use case. Check if we can use it.
+///
+/// For more details refer to: <https://github.com/bytecodealliance/wasmtime/blob/v27.0.0/src/commands/serve.rs#L641>
+fn use_pooling_allocator_by_default() -> Result<Option<bool>> {
```

Comment:
**nit:** I know wasmtime uses `Result<Option<bool>>`, but I don't really see why. In any case, I think `Result<bool>` is enough for our use case. 
```suggestion
fn use_pooling_allocator_by_default() -> Result<bool> {
```